doc/help/elektra-highlevel-gen.md

```diff
@@ -0,0 +1,415 @@
+# elektra-highlevel-gen(7) -- High-level API code-generation advanced features
+
+This document focuses on the advanced features of the high-level API code-generator template. We assume you already familiarized yourself
+with the basic features explained in [`kdb-gen-highlevel(1)`](kdb-gen-highlevel.md).
+
+## Configuration Options
+
+The parameters that are relevant to the concepts described here are (for the rest see [`kdb-gen-highlevel(1)](kdb-gen-highlevel.md)):
+
+- `enumConv`: allowed values: `strcmp`, `trie`, `default` (default)
+- `specLocation`: allowed values: `embedded` (default), `external`
+- `defaultsHandling`: allowed values: `embedded` (default), `speconly`
+- `specValidation`: allowed values: `none` (default), `minimal`
+
+The `enumConv` option is described [below](#enums).
+
+With `specLocation` and `defaultsHandling` you can configure what data is embedded into your application's binary. The default option for
```

Comment:
I am still not happy with this here. You describe a few scenarios but only write little about what these options actually do. Furthermore you describe the two options together as if they were one option. The tutorial also makes very clear (what was to me clear before) that it would be much better if we simply have one option for these selected scenarios.

But I also see that you really tried hard to make the tutorial nice, so if it is too much for you, you can leave it. I'll then read the docu again and try to give you better input how to describe the config options. The tutorial suggests to use these options like a single option anyway.

Reply:
> but only write little about what these options actually do.

I described everything these options do. `specLocation` switches between the embedded spec and the additional quickdump file. `defaultsHandling` enables/disables embedded default values.

> The tutorial also makes very clear (what was to me clear before) that it would be much better if we simply have one option for these selected scenarios.

I will see, if I find a sensible way to unify them.

Reply:
> I described everything these options do. specLocation switches between the embedded spec and the additional quickdump file. defaultsHandling enables/disables embedded default values.

These sentence are quite descriptive but you need to clarify "embedded spec" and "default values". Ideally this should be the same ("default values" a subset of "embedded spec").

> I will see, if I find a sensible way to unify them.

:sparkling_heart: